### PR TITLE
[UI] Add chart depth

### PR DESCRIPTION
**Pull request by dawidwoz**
# Problem 

Currently, the view is fixed on the chart depth. It means the chart can be very crowded. 
image

# Solution
We can enable a user to choose how many depth levels they want to see. We predefined six options: Full (current behaviour), one, two, three, four, and five. 

image

The change is meant to make the visualisation less crowded and help the user to find the right level of details they want to see. 

There are some examples. 

"I want to see the same structure as it would be in File Explorer" - level 1
image

"I want to see if the object I'm looking for is here or on the next level" - level 2
image

"I want to see where the actual files are, not just the folders making the structure" - going up in the levels. In this case - level 4

image

### CODE REVIEW

File: src/components/Options.tsx [hunk 1]
```diff
@@ -142,6 +146,18 @@ export const Options = memo(function Options() {
             iconMap={sizeMetricIcons}
           />
         </fieldset>
+        <fieldset className="rounded-lg border p-2">
+          <legend className="card__title ml-1.5 justify-start gap-2">
+            <Icon path={mdiViewModule} size="1.25em" />
+            Chart depth
```
Comment (joglr) [inline src/components/Options.tsx:152]: Could we simplify this to just be "Depth", to align with the rest of the options?

**Review by joglr — APPROVED** [review]
Hi Dawid, this is a really nice feature, that also helps with rendering larger repositories. I have a left one suggestion you might consider.

File: src/components/Options.tsx (hunk 1 above)
Comment (dawidwoz) [reply src/components/Options.tsx]: Good idea! Done! 🙂

**Comment by dawidwoz** [comment]
@joglr I don't have the right to merge PRs, so feel free to merge it when possible 🙂

**Pull request merged**